### ISSUE TITLE
CUDA error with batched solvers

### ISSUE THREAD

**Issue by tpadioleau** [ginkgo-project/ginkgo#1576]
Using Ginkgo batched solvers I sometimes get the following errors

```bash
terminate called after throwing an instance of 'gko::CudaError'
  what():  /tmp/soft/spack-stage/spack-stage-ginkgo-1.7.0-a6wuqlgaxa2t5vkyzdapmgicl73ujdjd/spack-src/cuda/base/executor.cpp:203: synchronize: cudaErrorMisalignedAddress: misaligned address
Abandon
```
This error appears on Ginkgo 1.7.0 release on a V100 GPU with CUDA 12.2.1.

You can find a minimal reproducer in this repository https://github.com/Geoflow/gko_minreproducer/tree/minimal-reproducer. In my case I got the error with a batch size of 1. Am I doing something wrong ?

cc @Geoflow

**Comment by tpadioleau** [comment]
thank you very much for the quick fix!

**Comment by tpadioleau** [comment]
I have 2 more questions:
1. In the current 1.7.0 version, is there a trick to make it work ? Changing precision, integer type ?
2. Do you plan a patch for version 1.7 or provide the bug fix when releasing 1.8 ?

**Comment by pratikvn** [comment]
This will definitely be fixed in 1.8.0. But one question from my side would be if it would be possible for you to depend on develop rather on a specific version ? Our `develop` is interface stable and will always be API compliant for each major version. For example, we may break interface in 2.x.x, but `develop` should not break interface with any 1.x.x (unless we believe the behaviour itself was flawed, and we strongly believe no user should be using it).

We plan to release 1.8.0 soon, probably in the next few weeks, so I dont think we will patch 1.7.

**Comment by tpadioleau** [comment]
I think it should be ok for us to wait 1.8.0. I would prefer to avoid the develop branch in case users notice different behaviour in the results of simulations.
